tests/ga4gh/drs/endpoints/test_service_info.py

```diff
@@ -0,0 +1,114 @@
+    'keys': [('id', 1)]
+}
+COLLECTION_CONFIG = {
+    'indexes': [INDEX_CONFIG],
+}
+DB_CONFIG = {
+    'collections': {
+        'objects': COLLECTION_CONFIG,
+        'service_info': COLLECTION_CONFIG,
+    },
+}
+MONGO_CONFIG = {
+    'host': 'mongodb',
+    'port': 27017,
+    'dbs': {
+        'drsStore': DB_CONFIG,
+    },
+}
+SERVICE_INFO_CONFIG = {
+    "contactUrl": "mailto:support@example.com",
+    "createdAt": "2019-06-04T12:58:19Z",
+    "description": "This service provides...",
+    "documentationUrl": "https://docs.myservice.example.com",
+    "environment": "test",
+    "id": "org.ga4gh.myservice",
+    "name": "My project",
+    "organization": {
+        "name": "My organization",
+        "url": "https://example.com"
+    },
+    "type": {
+        "artifact": "beacon",
+        "group": "org.ga4gh",
+        "version": "1.0.0"
+    },
+    "updatedAt": "2019-06-04T12:58:19Z",
+    "version": "1.0.0"
+}
+ENDPOINT_CONFIG = {
+    "objects": {
+        "id_charset": 'string.digits',
+        "id_length": 6
+    },
+    "access_methods": {
+        "id_charset": 'string.digits',
+        "id_length": 6
+    },
+    "service_info": deepcopy(SERVICE_INFO_CONFIG),
+    "url_prefix": "http",
+    "external_host": "1.2.3.4",
+    "external_port": 8080,
+    "api_path": "ga4gh/drs/v1"
+}
+
+
+def test_get_service_info():
+    """Test for getting correct service info config (new config added)."""
+    app = Flask(__name__)
+    app.config['FOCA'] = Config(
+        db=MongoConfig(**MONGO_CONFIG),
+        endpoints=ENDPOINT_CONFIG,
+    )
+
+    app.config['FOCA'].db.dbs['drsStore'].collections['service_info'] \
+        .client = mongomock.MongoClient().db.collection
+
+    with app.app_context():
+        service_info = RegisterServiceInfo()
+        service_info.set_service_info_from_config()
+
+    with app.app_context():
+        get_service_info = RegisterServiceInfo().get_service_info()
+        assert get_service_info == SERVICE_INFO_CONFIG
+
+
+def test_get_service_info_duplicatekey():
+    """Test for duplicated service info config."""
+    app = Flask(__name__)
+    app.config['FOCA'] = Config(
+        db=MongoConfig(**MONGO_CONFIG),
+        endpoints=ENDPOINT_CONFIG,
+    )
+
+    app.config['FOCA'].db.dbs['drsStore'].collections['service_info'] \
+        .client = mongomock.MongoClient().db.collection
+    mock = MagicMock(side_effect=[DuplicateKeyError(''), None])
+    app.config['FOCA'].db.dbs['drsStore'] \
+        .collections['service_info'].client.insert_one = mock
+    mock_db_call = MagicMock(name="Find_Obj")
+    mock_db_call.return_value.sort.return_value \
+        .limit.return_value.next.return_value = deepcopy(SERVICE_INFO_CONFIG)
+    app.config['FOCA'].db.dbs['drsStore'] \
+        .collections['service_info'].client.find = mock_db_call
+    with app.app_context():
+        get_service_info = RegisterServiceInfo().get_service_info()
+        assert get_service_info == SERVICE_INFO_CONFIG
```

Comment:
Please add the missing tests to get full coverage so that the checks pass.

If the module to be tested contains code that is not used as long as `POST /service-info` is not implemented (I didn't check that), it's best to remove the corresponding functions/methods from the `drs_filer.ga4gh.drs.endpoints.service_info` module and add them (together with  the tests) when you implement the `POST` endpoint

Reply:
I just checked and it looks like `set_service_info_from_config()` is used and so it should be fully tested in this PR. Lines not covered by tests in this method are lines 83, 84, 88 and 93 from module `drs_filer.ga4gh.drs.endpoints.service_info`. See here to see lines not covered by tests: https://coveralls.io/builds/39201369/source?filename=drs_filer%2Fga4gh%2Fdrs%2Fendpoints%2Fservice_info.py#L83

On the other hand, `set_service_info_from_app_context()` is only used in the `POST` endpoint controller to be implemented later so that metod and dependent methods `_upsert_service_info()` and `_get_headers()` can be removed from the `drs_filer.ga4gh.drs.endpoints.service_info` module for now and then of course do not need to be tested. You can then add back the three methods and corresponding tests in the PR for the `POST /service-info` endpoint.